Add GravityToggleStyle

djalfirevic opened the pull request.

djalfirevic committed:
Add GravityToggleStyle

djalfirevic committed:
Cleanup

djalfirevic commented on Sources/Props/Toggle/GravityToggleStyle.swift:
```diff
@@ -0,0 +1,68 @@
+    // MARK: - ToggleStyle
+    public func makeBody(configuration: Self.Configuration) -> some View {
+        GeometryReader { geometry in
+            let offset = geometry.size.width/2 - 3 * geometry.size.height/2 + 10
```
Maybe you have some better math for this. I don't like this magic 10, but it seems that it's working 🙏

qmchenry replied:
It's looking great! I recommend making the size fixed, roughly the size of the built-in toggle, but maybe a little longer to give the effect a little more oomph. If you fix the size, you shouldn't need geometry reading which will simplify, you can decide on the size, then know how big the magnitude of the animation. 

Also as a UX thing, it'd be great if the whole thing was tappable, not just the circle.

djalfirevic committed:
Update tappable area and set fixed frame

djalfirevic committed:
Merge branch 'main' of https://github.com/SmallPlanetLabs/Props into feature/gravity_toggle_style

qmchenry committed:
Add spacing to toggle example helper, fix readme content

djalfirevic committed:
Add secondary color

qmchenry committed:
Update gravity toggle style

Add enum reference
Change color handling for dark mode -- opacity doesn't work in dark mode
Check out `tintOrShade()` on Color which also has a Prop card under
the Colors category to show how it works.

qmchenry commented:
Check out my changes and see if you are into them. Let's add enum types for all new styles while we're implementing them plus some documentation following this example.

qmchenry committed:
Update documentation string for enum type

qmchenry commented on Catalog/Shared/Toggles/ToggleStyles.swift:
```diff
@@ -46,12 +47,28 @@ struct SunMoonToggleSample: PropSampleable {
+    let name = "gravity toggle style"
+    let notes: String? = ".toggleStyle(.gravity)"
+    let source = "Custom"
+    let keywords = "toggle custom props animated gravity"
```
'props" here was "third party" which referenced the sun/moon toggle above which is literally a 3rd party SPM imported into the catalog. Let's keep it consistent when we're using our props. Keep an eye on mine, too, as I'm likely to do this, too. :D

djalfirevic replied:
Oh right! I see. We should also change the source to be "Props" and not "Custom" then, to be consistent with others?

I've used your changes and applied to that other PR, but we'll check it out after this one ;)

djalfirevic committed:
Source: Props

djalfirevic committed:
Use Custom instead of Props

qmchenry committed:
Rearrange frame modifier on gravity toggle
